edb/pgsql/compiler/relgen.py

```diff
@@ -1954,18 +1956,19 @@ def _compile_func_epilogue(
     expr = ir_set.expr
     assert isinstance(expr, irast.FunctionCall)
 
-    if (ctx.volatility_ref is not None and
-            ctx.volatility_ref is not context.NO_VOLATILITY and
-            expr.volatility is qltypes.Volatility.VOLATILE):
-        # Apply the volatility reference.
-        # See the comment in process_set_as_subquery().
-        func_rel.where_clause = astutils.extend_binop(
-            func_rel.where_clause,
-            pgast.NullTest(
-                arg=ctx.volatility_ref,
-                negated=True,
+    if expr.volatility is qltypes.Volatility.VOLATILE:
+        for ref in ctx.volatility_ref:
+            if isinstance(ref, context.NoVolatilitySentinel):
```

Comment:
I didn't like the `NO_VOLATILITY` hack very much before, but in this formulation I like it even less.  Perhaps we could track the `INSERT` subject `path_id` instead and condition computable correlation in `process_set_as_subquery` on that instead.

Reply:
Yeah, I think that should work.